ldms/src/core/ldms_xprt.c

```diff
@@ -1678,7 +1678,7 @@ static int __process_dir_set_info(struct ldms_set *lset, enum ldms_dir_type type
 				ldms_dir_set_t dset, json_entity_t info_list)
 {
 	json_entity_t info_entity, e;
-	int j, rc;
+	int j, rc = 0;
```

Comment:
@baallan I think we don't need to initialize the `rc` here. The function will assign a value in case of an error before returning it.

Reply:
@nichamon if lset is not null but info_list is empty, rc is uninitialized unless we set it.
At the scope of this function, there is no assurance that info_list is populated, and gcc warns that rc can be used uninit at 1733.
If compiling such that warnings are promoted to error, this bombs the build.